### PR TITLE
Add info about indexed db ordering

### PR DESCRIPTION
**Pull request by SUPERCILEX**
Hey @puf @samtstern,
People have been confused in the past by db ordering with indexed data so this PR should clear things up.

#537 and another one I can't think of right now.

### CODE REVIEW

File: database/README.md [hunk 1]
```diff
@@ -301,3 +302,30 @@ new FirebaseIndexListAdapter<Chat>(this, Chat.class, android.R.layout.two_line_l
...
 
 `keyRef` is the location of your keys, and `dataRef` is the location of your data.
+
+### A note on ordering
+
+The order in which your receive your data depends on the order from `keyRef`, not `dataRef`:
+```json
+{
+  "data": {
+    // This order doesn't matter, the order is taken from keys/(user1 or user2).
```
Comment (SUPERCILEX) [inline database/README.md]: I'm not sure what to do about these, I want to add comments but json doesn't like that.

Comment (samtstern) [inline database/README.md]: Use "```javascript```" instead

Comment (SUPERCILEX) [inline database/README.md]: @samtstern Whoa, nice trick! 😄

**Comment by samtstern** [comment]
Squashing and merging.

**Pull request merged**